src/Endatix.Infrastructure/Features/WebHooks/WebHookServer.cs

```diff
@@ -0,0 +1,84 @@
+using System.Text;
+using System.Text.Json;
+using Endatix.Core.Features.WebHooks;
+using Microsoft.Extensions.Logging;
+using Polly.Timeout;
+
+namespace Endatix.Infrastructure.Features.WebHooks;
+
+/// <summary>
+/// Represents a server for firing WebHooks.
+/// </summary>
+internal class WebHookServer(HttpClient httpClient, ILogger<WebHookServer> logger)
+{
+    /// <summary>
+    /// Fires a WebHook asynchronously.
+    /// </summary>
+    /// <typeparam name="T">The type of the payload carried by the WebHook message.</typeparam>
+    /// <param name="message">The WebHook message to be sent.</param>
+    /// <param name="instructions">Properties for the WebHook operation, including the destination URI.</param>
+    /// <param name="token">A cancellation token.</param>
+    /// <returns>A task that represents the asynchronous operation. The task result indicates if the WebHook was successfully processed.</returns>
+    internal async Task<bool> FireWebHookAsync<T>(WebHookMessage<T> message, TaskInstructions instructions, CancellationToken token)
+    {
+        var isSuccess = false;
+        try
+        {
```

Comment:
I get a bit confused here. This try..catch handles TaskCanceledException and TimeoutRejectedException. If I'm not wrong, TaskCanceledException seem to be thrown only by the httpClient.SendAsync call. However, for the other I'm not sure.
It may be clearer if the try..catch is just around the specific code that is expected to throw a certain type of exception.
It seems to me that the code before SendAsync doesn't really need to be inside the try..catch.

Reply:
Hey, good question. It is the SendAsync indeed, which throws HttpExceptions of all sorts. So I kept the try/catch as close to the `await httpClient.SendAsync(request, token);` call as possible. I still kept the request orchestration code part of the try as it's closely related and we want full exception detection for any failed WebHook + trace of it in the logs. 

I have done extensive testing and decided to add these two exceptions as special ones as they are very common. Here is explanation:

`TimeoutRejectedException` - this is when a single request times out (in this case more than 10s.)
`TaskCanceledException` - the resilience pipeline will throw this when the entire execution hits the limit for a given webhook (in this case is either 120 sec or 5 failed attempts)
`Exception` - anything which is not the 2 expected failures

I am not re-throwing as the pipeline already has an interceptor for these requests and will retry in any of these 3 has been encountered (or the webhook receiving server responds with anything but 200 OK code)

```c#
typeof(TimeoutRejectedException),
typeof(SocketException),
typeof(HttpRequestException),

```

Will add comment to the exceptions, so it's easier for the future reader to understand

Reply:
Comments were added:
![image](https://github.com/user-attachments/assets/1b391990-c366-4589-8be5-263eefc5c36d)

Reply:
I understand, thanks for the explanation and adding the details in the comments.
Personally I would have put the try...catch just around the SendAsync because this is where the exceptions could come from (this is what I understand).
Of course, this is subjective and the current solution is absolutely fine.